readthedocs/ssh/mixins.py

```diff
@@ -0,0 +1,13 @@
+# -*- coding: utf-8 -*-
+from __future__ import division, print_function, unicode_literals
+
+from .keys import generate_ssh_pair_keys
+
+
+class SSHKeyGenMixin(object):
+
+    def generate_keys(self, commit=True):
+        """Generate random SSH pair keys."""
+        self.private_key, self.public_key = generate_ssh_pair_keys()
+        if commit:
+            self.save()
```

Comment:
Shouldn't this be a classmethod? Also, would this be better as a manager method, similar to `create()`?

Reply:
Mmm. I think it can't be a classmethod since we need the instance to save the just generated keys.

This is used in two different cases:

- when a SSHKey object is created and _not_ `private_key` is passed
- when an Organization (from corporate) is created and `public_key` is empty

We could define it in the manager as `SSHKeyQuerySet.generate()` but this will only cover the first case and we will need the method defined in this mixin copied into the `Organization` model from corporate.

Does it make sense to you?

----

Another side-topic related to this is that in the Organization case, we might want to adapt it since we are saving `public_key` and `private_key` inside the Organization model, which only allows us to have just one private key (but we can't have a `ForeignKey` to `Organization` in the `SSHKey` because this models lives in .org) --probably worth to do it in another PR if we decide to do it